packages/x-date-pickers/vitest.config.jsdom.mts

```diff
@@ -0,0 +1,33 @@
+import { mergeConfig } from 'vitest/config';
+// eslint-disable-next-line import/no-relative-packages
+import filterReplacePlugin from '../../test/vite-plugin-filter-replace.mts';
+import sharedConfig from '../../vitest.shared.mts';
+import { getTestName } from '../../scripts/getTestName.mts';
+
+export const filterReplace = filterReplacePlugin(
```

Comment:
I was under the impression we could solve this with vitest mocking. What's preventing us from doing so?

Reply:
This is too simple, it works for jsdom but not browser mode, which is the main reason we cant do the direct mocking.

Reply:
We can't use alias because it changes imports for ALL files.

We can't use `vi.mock` because the targets for browser and jsdom are different.

I should have kept more tabs on what I tried, but in general, I tried many things that worked only in one case but not the other no matter what I changed. 

The `plugin` way works for both cases only in the specified files. A bit annoying, but it works.

Reply:
yes, realized this quickly as well. But why doesn't this plugin I proposed not work in browser mode?

Reply:
Looks like it works for me, but I did have to exclude date-fns and date-fns-v2 from dependency optimization: https://stackblitz.com/edit/sb1-mjbqjjpy.

Reply:
Coming from webpack, I love the rollup/vite plugin system so much